deployment/spring-boot/pom.xml

```diff
@@ -64,6 +63,19 @@
       <scope>test</scope>
     </dependency>
 
+    <!-- needed for JUnit4 execution -->
+    <dependency>
+      <groupId>org.junit.vintage</groupId>
+      <artifactId>junit-vintage-engine</artifactId>
+      <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>org.hamcrest</groupId>
+          <artifactId>hamcrest-core</artifactId>
+        </exclusion>
+      </exclusions>
```

Comment:
Do we really need to exclude `hamcrest-core` here? I think it makes the example more complex for the user. 

Even though `junit-vintage-engine` overrides `hamcrest-core`, which is transitively pulled in via `spring-boot-starter-test`, I think it is not a big deal since both dependencies define with 2.2 the same version. Also, I would assume that the Spring Boot team takes care that both dependencies can be used together without further configuration since they define the dependencies in their `spring-boot-dependencies` bom.

Reply:
I took the config for using vintage junit from https://github.com/camunda/camunda-bpm-platform/blob/master/spring-boot-starter/starter-test/pom.xml, but you are right that a smaller pom is probably the better way for this example. 👍

Reply:
I haven't checked it, but I guess that the dependency tree in the `pom.xml` file you mentioned differs, so it might make sense to exclude the dependency based on the maven project hierarchy. However, in our case here, we should focus on keeping the example as simple as possible. 🙂